crates/starknet-os/src/starknet/core/os/contract_class/compiled_class_hash_objects.rs

```diff
@@ -35,6 +37,25 @@ impl BytecodeSegmentStructureImpl {
             BytecodeSegmentStructureImpl::Leaf(leaf) => leaf.add_bytecode_with_skipped_segments(data),
         }
     }
+
+    pub fn hash(&self) -> Result<FieldElement, SyscallExecutionError> {
+        let ret = match self {
+            BytecodeSegmentStructureImpl::SegmentedNode(node) => node.hash()?,
+            BytecodeSegmentStructureImpl::Leaf(leaf) => {
```

Comment:
Implement `hash()` for `BytecodeLeaf` instead of here to make this part more readable.